server/neptune/workflows/internal/deploy/revision/revision.go

```diff
@@ -77,25 +79,31 @@ func (n *Receiver) Receive(c workflow.ReceiveChannel, more bool) {
 		logger.Error(ctx, "generating deployment id", "err", err)
 	}
 
+	deploymentInfo := terraform.DeploymentInfo{
+		ID:             id,
+		Root:           root,
+		Revision:       request.Revision,
+		Repo:           repo,
+		InitiatingUser: initiatingUser,
+		Tags:           request.Tags,
+	}
+	// Do not push a duplicate/in-progress manual deployment to the queue
+	if root.Trigger == activity.ManualTrigger && (n.queue.Contains(deploymentInfo) || n.isInProgress(deploymentInfo)) {
```

Comment:
the deployment ID would be different and so could the initiating user. We should just be checking the revision.  I think `Contains` is a bit too specific of an implementation.  Considering just exposing a Scan method which returns a copy of the queue as a list (sorted by priority).